var/spack/repos/builtin/packages/jellyfish/package.py

```diff
@@ -18,7 +18,27 @@ class Jellyfish(AutotoolsPackage):
     version('1.1.11', sha256='496645d96b08ba35db1f856d857a159798c73cbc1eccb852ef1b253d1678c8e2',
             url='https://www.cbcb.umd.edu/software/jellyfish/jellyfish-1.1.11.tar.gz')
 
+    variant('pybind', default=False, description='Enable python binding')
+    variant('rubybind', default=False, description='Enable ruby binding')
+    variant('perlbind', default=False, description='Enable perl binding')
```

Comment:
I think we might be able to turn this into a multi-valued variant e.g. `languages=python`. Also, I think we want to have:
```python
extends('python', when='languages=python')
```
ans similar for other bindings.

Reply:
Hi

Yes, easy enough to turn into a multi-valued variant (used bindings rather than languages) 
```
variant(
        'bindings', default='none', description='Support for binding to Ruby, Python and Perl',
        values=('pybind', 'rubybind', 'perlbind', 'none'), multi=True
    )
```
although when I tested using extends instead of depends_on
```
extends('python', when='bindings=pybind')
```
received an AssertionError (quite possible I do not have something correct although the spec concretized)
```
==> Error: Class constructor failed for package 'builtin.jellyfish'.

Caused by:
AssertionError: 
  File "/home/stuart/code/teachers_github/spack/lib/spack/spack/repo.py", line 1125, in get
    return package_class(spec)
  File "/home/stuart/code/teachers_github/spack/lib/spack/spack/package_base.py", line 808, in __init__
    if self.is_extension:
  File "/home/stuart/code/teachers_github/spack/lib/spack/spack/package_base.py", line 1394, in is_extension
    return self.extendee_spec is not None
  File "/home/stuart/code/teachers_github/spack/lib/spack/spack/package_base.py", line 1363, in extendee_spec
    assert len(deps) == 1
```
Any suggestions welcome!

Kind regards

Reply:
Right. This is longstanding limitation of Spack, see https://github.com/spack/spack/issues/987, which I forgot about when writing the note above. 

Wondering if we can use a single-valued variant as a workaround for this. I assume that in your test above you specified more than one value for the `bindings` variant, right?

Reply:
Yes, I did a test with more than one value for `bindings`, I will do some testing with single-valued variants.

Reply:
Hi

So, subjected most aspects to some more testing and have reverted to the single-value variants with extends and only allowed one binding to be specified at a time.

Discovered that v1.1.11 does not support any bindings (would fail silently on build but I have added conflicts to make it more explicit) - started this because we had a package that depends on jellyfish@2.2.7 that requires a python binding and then another user wanted Kraken, which depends on jellyfish@1.1.11 (and I had lazily overridden the original jellyfish).

v2.2.7 only successfully built with python@2.7.x (our use case with the package that depends on jellyfish@2.2.7 already depended on python@:2.7.18 so it was already in the DAG most of the time I was using it).

Hope this helps!

Kind regards